packages/terra-worklist-data-grid/src/subcomponents/Row.jsx

```diff
@@ -0,0 +1,138 @@
+import rowShape from '../proptypes/rowShape';
+import WorklistDataGridPropTypes from '../proptypes/WorklistDataGridPropTypes';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The row's position in the Grid. This is zero based.
+   */
+  rowIndex: PropTypes.number,
+  /**
+   * String that specifies height of the row. Any valid CSS width value is accepted.
+  */
+  height: PropTypes.string,
+  /**
+   * The information about the Row.
+   */
+  row: rowShape.isRequired,
+  /**
+   * Boolean indicating whether or not the DataGrid should allow entire rows to be selectable. An additional column will be
+   * rendered to allow for row selection to occur.
+   */
+  hasSelectableRows: PropTypes.bool,
+  /**
+   * Callback function that will be called when a row is selected. Parameters: `onRowSelect(rowId)`
+   */
+  onRowSelect: PropTypes.func,
+  /**
+   * All columns currently displayed.
+   */
+  displayedColumns: PropTypes.arrayOf(WorklistDataGridPropTypes.columnShape),
+  /**
+   * Callback function that will be called when a cell in the row is selected.
+   */
+  onCellSelect: PropTypes.func,
+  /**
+   * A number indicating which column represents the row header. This is zero based.
+   */
+  rowHeaderIndex: PropTypes.number,
+  /**
+   * Id of the column in the row that is selected.
+   */
+  selectedCellColumnId: PropTypes.string,
+  /**
+   * Column index of the column in that has focus.
+   */
+  tabStopColumnIndex: PropTypes.number,
+
+};
+
+const defaultProps = {
+  hasSelectableRows: false,
+};
+
+function Row(props) {
+  const {
+    rowIndex,
+    height,
+    hasSelectableRows,
+    row,
+    displayedColumns,
+    rowHeaderIndex,
+    onRowSelect,
+    onCellSelect,
+    selectedCellColumnId,
+    tabStopColumnIndex,
+  } = props;
+
+  const theme = useContext(ThemeContext);
+
+  const handleCellSelect = (rowIdColId, coordinates) => {
+    if (hasSelectableRows) {
+      if (onRowSelect) {
+        onRowSelect(rowIdColId.rowId, rowIndex, coordinates);
+      }
+    } else if (onCellSelect) {
+      onCellSelect(rowIdColId, coordinates);
+    }
+  };
+
+  const getCellData = (cellRowIndex, cellColumnIndex, cell, rowId) => {
+    const columnId = displayedColumns[cellColumnIndex].id;
+    const isRowHeader = cellColumnIndex === (rowHeaderIndex + hasSelectableRows ? 1 : 0);
+
+    return (
+      <Cell
+        rowId={rowId}
+        columnId={columnId}
+        rowIndex={cellRowIndex}
+        columnIndex={cellColumnIndex}
+        key={`${rowId}_${columnId}`}
+        isTabStop={tabStopColumnIndex === cellColumnIndex}
+        isSelected={!hasSelectableRows && selectedCellColumnId === columnId}
+        cell={cell}
+        isRowHeader={isRowHeader}
+        onCellSelect={handleCellSelect}
+      />
+    );
+  };
+
+  const columnIndexOffSet = hasSelectableRows ? 1 : 0;
+  return (
+    <tr
+      className={cx([row.isSelected ? 'row-selected' : 'worklist-data-grid-row', theme.className])}
```

Comment:
Wouldn't every row be a worklsit-data-grid-row?  You would just add a "selected" class if the row is selected.  However, I would expect the worklist-data-grid-row class to always be there.

Reply:
We did not want the nth-child styling from the worklist-data-grid-row when the row was selected. Right now, it is just background color so we could get away with it but if something gets added to that style, the selected appearance may be affected. If there is something to be addressed we can address this on the row selection Jira: https://jira2.cerner.com/browse/UXPLATFORM-9043

Reply:
Ok.  Based on what I understand from your explanation, that could be accounted for pretty easily in CSS.

Reply:
I added this as a to do on the row selection Jira.